src/integration/java/io/pinecone/integration/controlPlane/pod/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -1,51 +1,181 @@
+        } catch (PineconeUnmappedHttpException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("dimension: invalid value"));
+        }
+    }
+
+    @Test
+    public void createIndexWithInvalidPods() {
+        CreateIndexRequestSpecPod podSpec =
+                new CreateIndexRequestSpecPod().environment(environment).pods(-1).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-pods")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("pods"));
+            assertTrue(expected.getLocalizedMessage().contains("must be greater than 0"));
+        }
+    }
+
+    @Test
+    public void createIndexWithInvalidReplicas() {
+        CreateIndexRequestSpecPod podSpec =
+                new CreateIndexRequestSpecPod().environment(environment).pods(1).replicas(-1).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-replicas")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("replicas"));
+            assertTrue(expected.getLocalizedMessage().contains("must be greater than 0"));
+        }
+    }
+
+    @Test
+    public void createIndexWithInvalidPodsToShards() {
+        CreateIndexRequestSpecPod podSpec =
+                new CreateIndexRequestSpecPod().environment(environment).pods(5).replicas(2).shards(2).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-shards")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("total pods must be divisible by number of shards"));
+        }
```

Comment:
Maybe worth adding a test to confirm an invalid similarity metric also raises an error? Just since you're doing tests for all the other index-creation params

Reply:
The argument type for `metric` is an enum: `IndexMetric`. Users are not able to pass `String` values or other values outside of this enum. I can add a test / some asserts around passing different similarity metrics though, or leaving it empty.

Reply:
ah TIL!